pkg/cluster/cluster.go

```diff
@@ -0,0 +1,72 @@
+package cluster
+
+import (
+	"github.com/ministryofjustice/cloud-platform-cli/pkg/client"
+	v1 "k8s.io/api/core/v1"
+)
+
+// Cluster represents useful values and object in a Kubernetes cluster
+type Cluster struct {
+	Name       string
+	Nodes      []v1.Node
+	Pods       []v1.Pod
+	OldestNode v1.Node
+	StuckPods  []v1.Pod
+}
+
+// Snapshot represents a snapshot of a Kubernetes cluster object
+type Snapshot struct {
+	Cluster Cluster
+}
+
+// New constructs a new Cluster object
+func New(c *client.Client) (*Cluster, error) {
```

Comment:
Is it possible to use a named function instead of doing the work inside the New(). Normally Constructors are used to zeroing or initializing

Reply:
Zeroing or initilizing in Go is done via c := Cluster{} (in this instance). I want to init with populated object, maybe I need to rename it.

Reply:
Changed the function name for readability